Initial update of dependencies to Jakarta EE 10 versions.

arjantijms opened the pull request:
First batch

arjantijms committed:
Update dependencies to Jakarta EE 10 versions.

First batch

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Second batch

arjantijms committed:
Third batch

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms commented:
Blocker: https://github.com/eclipse-ee4j/metro-wsit/issues/138

arjantijms committed:
Updated Servlet, Authorization and more. Added patches for Metro.

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms commented:
Blocker: https://hibernate.atlassian.net/browse/HV-1857

arjantijms committed:
Patch Hibernate Validator and integrate Sun EL 5.0.0-M1

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms commented:
Blocker: https://github.com/eclipse-ee4j/eclipselink/issues/1371

arjantijms committed:
Integrate WaSP 3.1.0-M1

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Faces and Mojarra 4.0.0 + Weld cleanup

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Integrated CDI 4/Weld 5

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms commented:
Blocker https://github.com/WASdev/standards.jsr352.jbatch/issues/74

arjantijms committed:
Integrate Batch 2.1.0-M1

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Improve logging of OSGi errors

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Integrate Standard Tags 3.0.0-SNAPSHOT API + Impl

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms commented:
Blocker https://github.com/eclipse-ee4j/tyrus/issues/762

jansupol commented on nucleus/parent/pom.xml:
```diff
@@ -83,41 +83,41 @@
-        <jax-rs-api.impl.version>3.0.0</jax-rs-api.impl.version>
-        <jersey.version>3.0.3</jersey.version>
+        <jax-rs-api.impl.version>3.1.0</jax-rs-api.impl.version>
+        <jersey.version>3.1.0-SNAPSHOT</jersey.version>
```
Jersey has 3.1.0-M1 in sonatype already

arjantijms replied:
Thanks, I'll try that one

arjantijms replied:
Unfortunately M1 is not strong enough, I get the following error again, which blocks the startup of the server completely:

```
3. java.lang.IllegalStateException: Unable to resolve
    org.glassfish.main.web.jersey-mvc-connector [31]
    missing requirement
        &(package = org.glassfish.jersey.server.mvc.jsp) (version >= 3.1.0) (!(version >= 4.0.0))
        caused by:
            Unable to resolve
                org.glassfish.jersey.ext.jersey-mvc-jsp [158]
                missing requirement
                    &(package = org.glassfish.jersey.server.mvc.internal) (version >= 3.1.0) (!(version >= 4.0.0))
                    caused by:
                        Unable to resolve
                            org.glassfish.jersey.ext.jersey-mvc [276]
                            missing requirement
                                &(package = jakarta.servlet) (version >= 5.0.0) (!(version >= 6.0.0)))]]
```

arjantijms replied:
I can try to hotpatch M1 though and try to make it as good as the current 3.1 branch.

jansupol replied:
Right, 3.1.0-M1 does not have the servlet OSGi headers set properly, yet

arjantijms replied:
M1 is probably also missing the CDI fixes. If I hot patch the archive, I get this:

```
 java.lang.NoSuchMethodError: 'jakarta.enterprise.inject.spi.InjectionTarget jakarta.enterprise.inject.spi.BeanManager.createInjectionTarget(jakarta.enterprise.inject.spi.AnnotatedType)'
	at org.glassfish.jersey.ext.cdi1x.servlet.internal.CdiExternalRequestScopeExtension.afterBeanDiscovery(CdiExternalRequestScopeExtension.java:72)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
```

It's probably better for GF to keep using the branch for now, and wait till there's a M2 with the necessary fixes.

arjantijms replied:
-m2 now available, and added. Thanks!

arjantijms committed:
Integrated WaSP further, disabled Krazo & Soteria

Tmp prevented exception when single initializer fails.

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Integrate Metro 4.0.0-M1 and XNL Bind RC3/M2

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms commented:
Blocker https://github.com/eclipse-ee4j/metro-wsit/issues/148

arjantijms committed:
Added santuario dependency and added new patch for Metro

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Updated CDI tests for CDI 4.0

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Integrated EclipseLink 4.0.0-M1 and Angus 1.0.0

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Small amount of cleaning before changes.

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Fix Jakarta Pages tests

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Added more info about what's happening in quicklook tests

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms commented:
Blocker: https://github.com/eclipse-ee4j/eclipselink/issues/1380

arjantijms committed:
Fixed QuickLook tests

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

pzygielo commented on appserver/tests/quicklook/QuickLook_Test_Instructions.html:
```diff
@@ -237,8 +237,8 @@ <H1 STYLE="margin-left: 0.83in"><FONT FACE="Trebuchet MS, sans-serif"><FONT SIZE
-quicklook/testng dir) for the profiles: testng_wd.xml for the Web
-profile, testng_gd.xml for Glassfish profile, testng_debug.xml for
+quicklook/testng dir) for the profiles: testng_web_profile.xml for the Web
+profile, ttestng_full_profile.xml for Glassfish full profile, testng_debug.xml for
```
extra `t` in `ttestng_full_profile.xml`?

arjantijms replied:
> ttestng_full_profile.xml

That's an amazing eagle eyed spot!

arjantijms committed:
Fix typo

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Fixed checkstyle warnings

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Tidying of poms

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Added snapshots module to build dependencies as part of the project.

This is used for dependencies we can't yet build and/or are not yet
released and are blocking.

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Merge remote-tracking branch 'origin/master' into ee10_dependencies

Conflicts:
	docs/performance-tuning-guide/src/main/jbake/content/tuning-glassfish.adoc

arjantijms committed:
Fix breakage of build

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Add jBatch snapshot

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Fix copy/paste error

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Run quicklook web twice

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Corrected typo

arjantijms committed:
Test web

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Add Hibernate Validator patch for Web Profile too.

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Added other jobs back

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
Update Jakarta Pages URI

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

arjantijms committed:
No TCK on profile oss-release

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

jGauravGupta reviewed (approved):
LGTM
